### PR TITLE
Add a flag to build the stdlib without COMPATIBILITY_OVERRIDE

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -406,6 +406,10 @@ option(SWIFT_ENABLE_EXPERIMENTAL_CONCURRENCY
   "Enable experimental Swift concurrency model"
   FALSE)
 
+option(SWIFT_ENABLE_COMPATIBILITY_OVERRIDES
```
Comment (compnerd) [inline CMakeLists.txt:409]: This seems to be specific to the stdlib would you mind just moving this to `stdlib/CMakeLists.txt` please?

**Review by compnerd — APPROVED** [review]
LGTM but please get @mikeash to also look at it before merging.

**Review by mikeash — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Pull request merged**